docs/blog/2017-07-19-creating-a-blog-with-gatsby/index.md

```diff
@@ -0,0 +1,486 @@
+        });
+
+      posts
+        .forEach(({ node }) => {
+          createPage({
+            path: node.frontmatter.path,
+            component: blogPostTemplate,
+            context: {} // additional data can be passed via context
+          });
+        });
+
+      return posts; // not strictly necessary, but continues the Promise chain with the sorted posts
+    });
+}
+```
+
+We've now tied into the Promise chain exposed by the `graphql` query. The actual posts are available via the path `result.data.allMarkdownRemark.edges`. Each edge contains an internal node, and this node holds the useful data that we will use to construct a page with Gatsby. Our GraphQL "shape" is directly reflected in this data object, so each property we pulled from that query will be available when we are querying in our GraphQL blog post template.
+
+The `createPage` API accepts an object which requires `path` and `component` properties to be defined, which we have done above. Additionally, an optional property `context` can be used to inject data and make it available to the blog post template component via injected props (log out props to see each available prop!). Each time we build with Gatsby, `createPage` will be called, and Gatsby will create a static HTML file of the path we specified in the post's frontmatter--the result of which will be our stringified and parsed React template injected with the data from our GraphQL query. Whoa, it's actually starting to come together!
+
+We can run `yarn develop` at this point, and then navigate to `http://localhost:8000/hello-world` to see our first blog post, which should look something like below:
+
+![My first blog post with Gatsby](./my-first-blog-post.png)
+
+At this point, we've created a single static blog post as an HTML file, which was created by a React component and several GraphQL queries. However, this isn't a blog! We can't expect our users to guess the path of each post, we need to have an index or listing page, where we display each blog post, a short snippet, and a link to the full blog post. Wouldn't you know it, we can do this incredibly easily with Gatsby, using a similar strategy as we used in our blog template, e.g. a React component and a GraphQL query.
+
+## Creating the Blog Listing
+
+I won't go into quite as much detail for this section, because we've already done something very similar for our blog template! Look at us, we're pro Gatsby-ers at this point!
+
+Gatsby has a standard for "listing pages," and they're placed in the root of our filesystem we specified in `gatsby-source-filesystem`, e.g. `src/pages/index.js`. So create that file if it does not exist, and let's get it working! Additionally note that any JavaScript files (typically React components!) will get a corresponding static HTML file when built. For instance, if we create `src/pages/tags.js`, the path `http://localhost:8000/tags.html` will be available within the browser and the statically generated site.
```

Comment:
tags.js will become /tags/index.html when when statically built but its browser path is just `/tags/` which is probably the least confusing way of describing it.